### DISCUSSION TITLE
Using ImGui::ColorConvertU32ToFloat4 in components.hpp templates

### DISCUSSION THREAD

**Discussion by T14D3** [YimMenu/YimMenu#1157] [Q&A]
While trying to make the general GUI look prettier and be more customizable, I've run into the issue of not knowing how I could use something like "ImGui::ColorConvertU32ToFloat4(some.config.section)" inside of any of the "template<>" of the "components.hpp" file.
Any insight on this is greatly appreciated.

Example:
```
template<ImVec2 size = ImVec2(0, 0), ImVec4 color = ImVec4(0.24f, 0.23f, 0.29f, 1.00f)>
		static void button(const std::string_view text, std::function<void()> cb)
		{
			if (button<size, color>(text))
			{
				g_fiber_pool->queue_job(cb);
			}
		}
```
Any attempt to replace `ImVec4(0.24f, 0.23f, 0.29f, 1.00f)` with some variation of `ImGui::ColorConvertU32ToFloat4(some.config.section)` refuses to compile

**Comment by maybegreat48** [comment]
Template parameters must be constexpr, so you have to create another function that takes the color as a normal parameter instead

Untested:
```c++
template<ImVec2 size = ImVec2(0, 0)>
static bool button(const std::string_view text, ImVec4 color)
{
	bool status = false;
	ImGui::PushStyleColor(ImGuiCol_Button, color);
	status = ImGui::Button(text.data(), size);
	ImGui::PopStyleColor(1);
	return status;
}

template<ImVec2 size = ImVec2(0, 0)>
static void button(const std::string_view text, std::function<void()> cb, ImVec4 color)
{
	if (button<size>(text, color))
	{
		g_fiber_pool->queue_job(cb);
	}
}
```

**Reply by Seanghost117** [reply]
With this make sure to give it a initial value or u might fuck up a lot of the project because you will need to manually add the default color or what not to every single button. What maybegreat suggested is more actionable for what you need, I opted to do it differently for the sake of being neat especially with all the buttons with call backs.

**Reply by maybegreat48** [reply]
> With this make sure to give it a initial value or u might fuck up a lot of the project

I meant that they should add these as new overloads, not replace the existing ones
